common/src/main/java/org/sitmun/authorization/service/ProxyConfigurationService.java

```
@@ -0,0 +1,133 @@
+		List<String> varyParameters = new ArrayList<String>();
+		addParameters(parameters, service.getParameters(), varyParameters);
+		
+		HttpSecurityDto security = null;
+		if (service.getPasswordSet()) {
+			security = HttpSecurityDto.builder()
+			  .type("http")
+			  .scheme("basic")
+			  .username(service.getUser())
+			  .password(service.getPassword())
+			  .build();
+		}
+		
+		return ConfigProxyDto.builder()
+		  .type(service.getType())
+		  .exp((new Date().getTime() / 1000) + responseValidityTime)
+		  .vary(varyParameters)
+		  .payload(OgcWmsPayloadDto.builder()
+		    .uri(service.getServiceURL())
+			.method(method)
+			.parameters(parameters)
+			.security(security)
+			.build())
+		  .build();
+	}
+	
+	private ConfigProxyDto getDatasourceConfiguration(DatabaseConnection databaseConnection, String sql) {
+		return ConfigProxyDto.builder()
+		  .type(CONNECTION_TYPE_KEY)
+		  .exp((new Date().getTime() / 1000) + responseValidityTime)
+		  .vary(new ArrayList<String>())
+		  .payload(DatasourcePayloadDto.builder()
+		    .uri(databaseConnection.getUrl())
+		    .user(databaseConnection.getUser())
+		    .password(databaseConnection.getPassword())
+		    .driver(databaseConnection.getDriver())
+		    .sql(sql)
+		    .build())
+		  .build();
+	}
+	
+	private void addParameters(Map<String, String> requestParameters, Set<ServiceParameter> serviceParameters, List<String> varyParameters) {
+		serviceParameters.forEach(p -> {
+			if(VARY_KEY.equalsIgnoreCase(p.getType())) {
+				varyParameters.add(p.getName());
+			} else if (!requestParameters.containsKey(p.getName())){
+				requestParameters.put(p.getName(), p.getValue());
+			}
+		});
+	}
+	
+	private String getSqlByTask(Task task) {
+		String sql = "";
+		Map<String, Object> taskParams = task.getProperties();
+		if(taskParams != null && taskParams.containsKey(SQL_COMMAND_KEY)) {
+			sql = (String)taskParams.get(SQL_COMMAND_KEY);
+		}
+		return sql;
```

What happens when the SQL is an empty string? The request is probably wrong. Fail fast, throw an exception that returns a 401.